tests/mpmc_channel.rs

```diff
@@ -138,14 +137,34 @@ macro_rules! gen_mpmc_tests {
                 assert!(receive_fut.as_mut().is_terminated());
             }
 
+            // A stream future shouldn't terminate until the stream
+            // terminates.
+            fn assert_receive_stream_done<FutureType, T>(
```

Comment:
Since that method is 99.9% of the one above, it seems like one should call the other, and then just add the `is_terminated()` line. I'm however not yet decided on a good name for both.

However if we rather test the `Stream` directly then via `StreamExt` then this most likely needs to stay anyway, and some things would need to get renamed (`FutureType` -> `StreamType`).